packages/@aws-cdk/aws-stepfunctions-tasks/lib/stepfunctions/start-execution.ts

```diff
@@ -0,0 +1,127 @@
+import * as iam from '@aws-cdk/aws-iam';
+import * as sfn from '@aws-cdk/aws-stepfunctions';
+import { Construct, Stack } from '@aws-cdk/core';
+import { integrationResourceArn, validatePatternSupported } from '../private/task-utils';
+
+/**
+ * Properties for StartExecution
+ */
+export interface StepFunctionsStartExecutionProps extends sfn.TaskStateBaseProps {
+  /**
+   * The Step Functions state machine to start the execution on.
+   */
+  readonly stateMachine: sfn.IStateMachine;
+
+  /**
+   * The JSON input for the execution, same as that of StartExecution.
+   *
+   * @see https://docs.aws.amazon.com/step-functions/latest/apireference/API_StartExecution.html
+   *
+   * @default - No input
+   */
+  readonly input?: sfn.TaskInput;
+
+  /**
+   * The name of the execution, same as that of StartExecution.
+   *
+   * @see https://docs.aws.amazon.com/step-functions/latest/apireference/API_StartExecution.html
+   *
+   * @default - None
+   */
+  readonly name?: string;
+}
+
+/**
+ * A Step Functions Task to call StartExecution on another state machine.
+ *
+ * It supports three service integration patterns: FIRE_AND_FORGET, SYNC and WAIT_FOR_TASK_TOKEN.
+ */
+export class StepFunctionsStartExecution extends sfn.TaskStateBase {
+  private static readonly SUPPORTED_INTEGRATION_PATTERNS = [
+    sfn.IntegrationPattern.REQUEST_RESPONSE,
+    sfn.IntegrationPattern.RUN_JOB,
+    sfn.IntegrationPattern.WAIT_FOR_TASK_TOKEN,
+  ];
+
+  protected readonly taskMetrics?: sfn.TaskMetricsConfig;
+  protected readonly taskPolicies?: iam.PolicyStatement[];
+
+  private readonly integrationPattern: sfn.IntegrationPattern;
+
+  constructor(scope: Construct, id: string, private readonly props: StepFunctionsStartExecutionProps) {
+    super(scope, id, props);
+
+    this.integrationPattern = props.integrationPattern || sfn.IntegrationPattern.REQUEST_RESPONSE;
+    validatePatternSupported(this.integrationPattern, StepFunctionsStartExecution.SUPPORTED_INTEGRATION_PATTERNS);
+
+    if (this.integrationPattern === sfn.IntegrationPattern.WAIT_FOR_TASK_TOKEN && !sfn.FieldUtils.containsTaskToken(props.input)) {
+      throw new Error('Task Token is required in `input` for callback. Use Context.taskToken to set the token.');
+    }
+
+    this.taskPolicies = this.createScopedAccessPolicy();
+  }
+
+  protected renderTask(): any {
+    return {
+      Resource:
+        integrationResourceArn('states', 'startExecution', this.integrationPattern) +
+        (this.integrationPattern === sfn.IntegrationPattern.RUN_JOB ? ':2' : ''),
```

Comment:
using `.sync:2` is a good default, not sure an option for `.sync` is necessary.

Reply:
Add a comment here on what the `:2` is or a link to documentation.

Prefer using string interpolation over string concatenation using `+`.